overlay/generic/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -30,73 +47,137 @@ dns_resolver2="8.8.4.4"
 declare -a states
 declare -a nics
 declare -a assigned
-declare -a DISK_LIST
+declare prmpt_str
+
+nicsup_done=0
 
-sigexit()
+fatal()
 {
 	echo
-	echo "System configuration has not been completed."
+	if [[ -n "$1" ]]; then
+		echo "ERROR: $1"
+	fi
+	echo
+	echo "System configuration failed, launching a shell."
 	echo "You must reboot to re-run system configuration."
-	exit 0
+	echo "A log of setup may be found in /var/log/prompt-config.log"
+	echo
+	/usr/bin/bash
 }
 
-#
-# Get the max. IP addr for the given field, based in the netmask.
-# That is, if netmask is 255, then its just the input field, otherwise its
-# the host portion of the netmask (e.g. netmask 224 -> 31).
-# Param 1 is the field and param 2 the mask for that field.
-#
-max_fld()
+
+sig_doshell()
 {
-	if [ $2 -eq 255 ]; then
-		fmax=$1
-	else
-		fmax=$((255 & ~$2))
-	fi
+	echo
+	echo
+	echo "Bringing up a shell.  When you are done in the shell hit ^D to"
+	echo "return to the system configuration tool."
+	echo
+
+	/usr/bin/bash
+
+	echo
+	echo "Resuming the system configuration tool."
+	echo
+	printf "$prmpt_str"
 }
 
-#
-# Converts an IP and netmask to a network
-# For example: 10.99.99.7 + 255.255.255.0 -> 10.99.99.0
-# Each field is in the net_a, net_b, net_c and net_d variables.
-# Also, host_addr stores the address of the host w/o the network number (e.g.
-# 7 in the 10.99.99.7 example above).  Also, max_host stores the max. host
-# number (e.g. 10.99.99.254 in the example above).
-#
-ip_netmask_to_network()
+ip_to_num()
 {
 	IP=$1
-	NETMASK=$2
 
 	OLDIFS=$IFS
 	IFS=.
 	set -- $IP
-	net_a=$1
-	net_b=$2
-	net_c=$3
-	net_d=$4
-	addr_d=$net_d
-
-	set -- $NETMASK
-
-	# Calculate the maximum host address
-	max_fld "$net_a" "$1"
-	max_a=$fmax
-	max_fld "$net_b" "$2"
-	max_b=$fmax
-	max_fld "$net_c" "$3"
-	max_c=$fmax
-	max_fld "$net_d" "$4"
-	max_d=$(expr $fmax - 1)
-	max_host="$max_a.$max_b.$max_c.$max_d"
-
-	net_a=$(($net_a & $1))
-	net_b=$(($net_b & $2))
-	net_c=$(($net_c & $3))
-	net_d=$(($net_d & $4))
-
-	host_addr=$(($addr_d & ~$4))
+	num_a=$(($1 << 24))
```

Comment:
in some functions `local` is used, in this one it is not.  all of these variables will be made global.